### PR TITLE
International address updates

### PR DESCRIPTION
**Pull request by bdorfman**
Changes non-US address forms to not require the State/County field.
Updates phone number validation to use country codes other than from the device's locale.

### CODE REVIEW

File: Stripe/STPPhoneNumberValidator.m [hunk 1]
```diff
@@ -12,37 +12,81 @@
...
+    NSString *countryCode = nillableCode;
+    if (!countryCode) {
+        countryCode = [[NSLocale autoupdatingCurrentLocale] objectForKey:NSLocaleCountryCode];
+    }
+    return countryCode;
+}
+                                                           
 + (BOOL)stringIsValidPartialPhoneNumber:(NSString *)string {
-    if (![self isUSLocale]) {
+    return [self stringIsValidPartialPhoneNumber:string forCountryCode:nil];
```
Comment (benzguo) [inline Stripe/STPPhoneNumberValidator.m]: I think `isUSLocale` is now unused and can be deleted

Comment (bdorfman) [inline Stripe/STPPhoneNumberValidator.m]: 👍

File: Stripe/STPAddress.m [hunk 2]
```diff
@@ -83,12 +83,12 @@ - (BOOL)containsRequiredFields:(STPBillingAddressFields)requiredFields {
 }
 
 - (BOOL)hasValidPostalAddress {
-    return self.line1.length > 0 &&
-    self.city.length > 0 &&
-    self.state.length > 0 &&
-    self.country.length > 0 &&
-    [STPPostalCodeValidator stringIsValidPostalCode:self.postalCode 
-                                        countryCode:self.country];
+    return (self.line1.length > 0
```
Comment (benzguo) [inline Stripe/STPAddress.m]: can we add a few tests for `hasValidPostalAddress` to `STPAddressTests`?

Comment (bdorfman) [inline Stripe/STPAddress.m]: I added a bunch of new checks to STPAddressTests. `containsRequiredFields:` with `Full` as the type calls through to hasValidPostalAddress so this should be covered well now.

**Pull request merged**